### PR TITLE
Some More Legacy Redirect Fixes

### PR DESCRIPTION
**Pull request by danielcliu-ifixit**
## Overview

Basically #529 had comments floating around it in and there were some more complaints to address.
This PR

1. Handles incorrect item types where they have spaces instead of "_" so things like `Parts/iPhone_11/Adhesive%20Strips` will now correctly reroute to `/Adhesives`. This might seem like an unnecessary fix but we are already handling things like `iPhone%2011` correctly so at least this stays consistent.
2. Redirect both  `/SSD_Enclosures` and `/Upgrade_Kits` to  `/Hard_Drives`.
3. Reorganize the `tool-collections` redirect generation so its more analogous to the parts one.

## QA
It should be pretty easy to test 1.) and 2.) above. 
`/Parts/HTC_Vive/I%2FO%20Board` seems like a decent route to test with.

Connects: #529

### CODE REVIEW

**Comment by deltuh-vee** [comment]
QA 🎬 
* Item types with spaces get redirected to their proper routes
* Both /SSD_Enclosures and /Upgrade_Kits links redirect to /Hard_Drives

File: frontend/next-config/redirects/part-collections.js [hunk 1]
```diff
@@ -27,9 +27,10 @@ function getLegacyPartItemTypeRedirects() {
 }
 
 function getRegexForTag(oldTag) {
-   return `(${escapeStringForRegex(
-      encodeURIComponent(destylizeURIComponent(oldTag))
-   )}|${escapeStringForRegex(oldTag)})`;
+   const encodedTag = encodeURIComponent(destylizeURIComponent(oldTag));
+   return encodedTag === oldTag
+      ? escapeStringForRegex(oldTag)
+      : `(${escapeStringForRegex(encodedTag)}|${escapeStringForRegex(oldTag)})`;
```
Comment (ardelato) [inline frontend/next-config/redirects/part-collections.js:33]: Wouldn't this regex cover both cases so the ternary operator would not be needed?
- `encodedTag` **equals** `oldTag`
- `encodedTag` **does not equal** `oldTag`

**Review by ardelato — APPROVED** [review]
CR 👍 , but question about the regex matching

File: frontend/next-config/redirects/part-collections.js (hunk 1 above)
Comment (danielcliu-ifixit) [reply frontend/next-config/redirects/part-collections.js:33]: Yes this is to reduce the regex where we can. While `(Hello|Hello)` would technically work, i think we should just reduce it to `Hello` where we can.

**Review by masonmcelvain — APPROVED** [review]
CR ✌🏻

**Pull request merged**